src/pages/ServiceGraph/SummaryPanelCommon.ts

```diff
@@ -29,11 +44,48 @@ export const updateHealth = (summaryTarget: any, stateSetter: (hs: HealthState)
   }
 };
 
-export const nodeData = (node: any) => {
+export const nodeData = (node: any): NodeData => {
   return {
     namespace: node.data('namespace'),
     app: node.data('app'),
     version: node.data('version'),
     workload: node.data('workload')
   };
 };
+
+export const nodeGlobalData = (node: any) => {
+  return node.cy().scratch(CytoscapeGlobalScratchNamespace) as CytoscapeGlobalScratchData;
+};
+
+export const getNodeMetrics = (
+  node: any,
+  props: SummaryPanelPropType,
+  filters: Array<string>,
+  includeIstio?: boolean,
+  byLabelsIn?: Array<string>
+) => {
+  // Determine if is by workload or by app
+  const data = nodeData(node);
+  const globalData = nodeGlobalData(node);
+  const isByApp = globalData.graphType === GraphType.APP && data.app !== 'unknown';
+
+  const options: MetricsOptions = {
+    queryTime: props.queryTime,
+    duration: props.duration,
+    step: props.step,
+    rateInterval: props.rateInterval,
+    includeIstio: includeIstio,
+    filters: filters,
+    byLabelsIn: byLabelsIn
+  };
+
+  if (globalData.versioned && data.version && data.version !== 'unknown') {
+    options.version = data.version;
```

Comment:
I thought I removed options.version (maybe only on backend, and it wasn't changed on frontend?)
Basically the idea is that a "versioned app" is actually a workload.

Reply:
> I thought I removed options.version (maybe only on backend, and it wasn't changed on frontend?)

Right, it wasn't changed in frontend and i think this is causing this one https://issues.jboss.org/browse/KIALI-1276



> A "versioned app" is not a workload - in the "versioned app" graph, for example, we don't show "workload name" in the node, we should "app" label with "version" label. Also, app/version pairs can span workloads (I can have workload "A-1" and "B-2" both labeled with "appA-versionV").

That's what i understood

Reply:
Ok, I talked with mazz and jay today, this needs a little more clarification and I asked this on istio-dev: https://groups.google.com/forum/#!topic/istio-dev/Ll9AxJjZYaY

My initial assumption was that the app-based graph was workload-based under the hood, then aggregated by app. But if "version" still plays a role then, you're probably right and we'll have to put "version" back in the query params server-side.